recipes/llvm-core/all/conanfile.py

```diff
@@ -1,429 +1,510 @@
-from conan.errors import ConanInvalidConfiguration
 from conan import ConanFile
-from conan.tools.build import cross_building
-from conan.tools.files import apply_conandata_patches, chdir, collect_libs, get, load, rename, replace_in_file, rm, rmdir, save
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os
+from conan.tools.build import check_min_cppstd, can_run, cross_building
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualRunEnv
+from conan.tools.files import (
+    apply_conandata_patches,
+    collect_libs,
+    get,
+    rmdir,
+    save,
+    copy,
+    export_conandata_patches,
+    rm,
+    rename,
+    replace_in_file
+)
+from conan.tools.microsoft import is_msvc
 from conan.tools.scm import Version
-from conans import CMake
-from collections import defaultdict
+
 import json
-import re
-import os.path
 import os
+from pathlib import Path
+import re
 import textwrap
 
-required_conan_version = ">=1.50.2" # Due to conan.tools.scm.Version
+
+required_conan_version = ">=1.62.0"
+
+# LLVM's default config is enable all targets, but end users can significantly reduce
+# build times for the package by disabling the ones they don't need with the corresponding option
+# `-o llvm-core/*:with_target_<target name in lower case>=False`
+LLVM_TARGETS = [
+    "AArch64",
+    "AMDGPU",
+    "ARM",
+    "AVR",
+    "BPF",
+    "Hexagon",
+    "Lanai",
+    "LoongArch",
+    "Mips",
+    "MSP430",
+    "NVPTX",
+    "PowerPC",
+    "RISCV",
+    "Sparc",
+    "SystemZ",
+    "VE",
+    "WebAssembly",
+    "X86",
+    "XCore"
+]
 
 
 class LLVMCoreConan(ConanFile):
-    name = 'llvm-core'
+    name = "llvm-core"
     description = (
-        'A toolkit for the construction of highly optimized compilers,'
-        'optimizers, and runtime environments.'
+        "A toolkit for the construction of highly optimized compilers,"
+        "optimizers, and runtime environments."
     )
-    license = 'Apache-2.0 WITH LLVM-exception'
-    topics = ('llvm', 'compiler')
-    homepage = 'https://llvm.org'
-    url = 'https://github.com/conan-io/conan-center-index'
-
-    settings = ('os', 'arch', 'compiler', 'build_type')
+    license = "LLVM-exception"
```

Comment:
This license change is wrong. Please undo

Reply:
Are you sure? This is the SPDX identifier for the LLVM license, according to https://spdx.org/licenses/LLVM-exception.html, and according to the [conan docs](https://docs.conan.io/2/reference/conanfile/attributes.html#license), "..it is strongly recommended that recipes of Open Source projects use [SPDX](https://spdx.dev/) identifiers from the [SPDX license list](https://spdx.org/licenses/)"

Reply:
The existing license was correct. See https://spdx.org/licenses/exceptions-index.html
> The SPDX License List includes a list of exceptions. These exceptions grant an exception to a license condition or additional permissions beyond those granted in a license; they are not stand-alone licenses. Exceptions are added to a license using the [License Expression](https://spdx.org/specifications) operator, "WITH".

Reply:
What you linked is just an exception identifier not a license identifier. @planetmarshall

Reply:
Reverted - thanks for the clarification.